tree.go

```diff
@@ -0,0 +1,62 @@
+package rsmt2d
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/NebulousLabs/merkletree"
+)
+
+var DefaultTreeCreator = &defaultTreeCreator{}
+
+type Tree interface {
+	Push(data []byte)
+	// TODO(ismail): is this general enough?
+	Prove(idx int) (merkleRoot []byte, proofSet [][]byte, proofIndex uint64, numLeaves uint64)
+	Root() []byte
+}
+
+type TreeCreator interface {
```

Comment:
I think TreeFactory would be a more conventional name?

Reply:
Yeah, maybe. The reason I didn't call this factory is that it currently is meant to create and return one concrete implementation of a `Tree` (`New` doesn't take in any params). Kind of a boring factory  (at least currently).

Reply:
Got rid of it in https://github.com/lazyledger/rsmt2d/pull/9/commits/51a97d9ef976d3c58b27de1c26678f1f4ac28a73